code/modules/vehicles/mecha/combat/greyscale/greyscale_constructor.dm

```diff
@@ -442,3 +450,29 @@ GLOBAL_LIST_INIT(greyscale_weapons_data, generate_greyscale_weapons_data())
 		if(slot == MECH_GREY_HEAD)
 			new_overlays += icon2appearance(SSgreyscale.GetColoredIconByType(initial(typepath.visor_config), selected_visor))
 	mech_view.overlays = new_overlays
+
+// MECH BUILDER LOCKING SYSTEM
+
+///variant of the mech builder with no cooldown, used in valhalla
+/obj/machinery/computer/mech_builder/valhalla
+	name = "valhalla mech computer"
+	desc = "A magic mech computer that can assemble new mechs almost instantly."
+	lock_timer = 5 SECONDS
```

Comment:
lumi told me to put a lower timer instead of using istype to ignore the lock completely because istype is a bad habit (this will also limit valhalla mechs to two but they will just have to destroy the one they currently have assembled to get a new one) this makes it so valhalla isnt full of leftover mechs filling the test zones